File: demo-community/domain/src/main/java/com/demo/domain/post/Post.java
package com.demo.domain.post;

import com.demo.domain.code.PostStatus;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.time.LocalDateTime;

@Entity
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "post_id")
    private Long id;

    private Long memberUid;

    private String title;

    @Column(columnDefinition = "text")
    private String content;

    @Enumerated(EnumType.STRING)
    private PostStatus status;

    private LocalDateTime publishedAt;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;

    protected Post() {}

    public Post(Long memberUid, String title, String content) {
        this.memberUid = memberUid;
        this.title = title;
        this.content = content;
    }
}

File: cinema-friends/domain/src/main/kotlin/com/cinemafriends/post/entity/Post.kt
package com.cinemafriends.post.entity

import java.time.LocalDateTime

class Post(
    id: Long? = null,
    val title: String,
    val content: String,
    val author: String,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now()
) {
    private val noneId = -9999L
    val id: Long = id ?: noneId
}